src/Connection/PhpiredisSocketConnection.php

```diff
@@ -308,7 +310,10 @@ public function connect()
     {
         if (parent::connect() && $this->initCommands) {
             foreach ($this->initCommands as $command) {
-                $this->executeCommand($command);
+                $response = $this->executeCommand($command);
+                if ($response instanceof ResponseInterface && $response instanceof ErrorResponseInterface) {
```

Comment:
Actually `Predis\Response\ErrorInterface` extends `Predis\Response\ResponseInterface`, there's no need for the first check.